core/src/main/java/com/bettercloud/bigtable/orm/Dao.java

```diff
@@ -1,13 +1,24 @@
 package com.bettercloud.bigtable.orm;
 
 import java.io.IOException;
+import java.util.Map;
 import java.util.Optional;
+import java.util.Set;
 
 public interface Dao<T extends Entity> {
 
+    @Deprecated
     <K extends Key<T>> Optional<T> get(final K key) throws IOException;
 
+    <K extends Key<T>> Map<K, T> getAll(final Set<K> keys) throws IOException;
+
+    @Deprecated
     <K extends Key<T>> T save(final K key, final T entity) throws IOException;
 
+    <K extends Key<T>> Map<K, T> saveAll(final Map<K, T> entities) throws IOException;
```

Comment:
I responded privately, but in the spirit of open-source I'll repeat myself here.

You mention a valid concern. I considered a List contract at first, but decided to go with a Map for a few reasons:

1) The Entity classes currently has no knowledge of their key. Introducing that functionality is a bigger overall change. The alternative would be to create a container with both the key and the value to be stored, but that defeats the purpose of preventing CPU/GC overhead.
2) Using a Map negates the chances for redundant writes (where a List could have multiple Entities with the same key).
3) The open-source interface shouldn't make too many assumptions about the use case. We may be using Kafka, which uses a List by its very nature, but should we dictate our interface design based on our own technology choices?

All of that said, the HBase API consumes a List of Put objects, so using a List instead of a Map would be consistent with their interface choice.

Version 2 is inevitable at some point in the future to remove the now-deprecated single-operation methods. I'd be more inclined to introduce that type of change in a breaking version.